clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -461,12 +461,27 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
   }
 
 private:
-  bool CheckSYCLType(QualType Ty, SourceRange Loc) {
+  /// Checks whenever it is legal or not to use particular type (\arg \c Ty)
+  /// within SYCL device code
+  ///
+  /// @param Ty Type to be checked
+  /// @param Loc SourceRange pointing to the expression which type is being
+  /// analyzed for checking of usage correctness
+  /// @param Decl The exact declaration which is being analyzed for checking of
```

Comment:
You say that `Decl` parameter is declaration but it has `DeclRefType` which means reference to declaration. Confusion.